server/filestore.go

```diff
@@ -3193,10 +3207,16 @@ func (fs *fileStore) NumPending(sseq uint64, filter string, lastPerSubject bool)
 			}
 			var smv StoreMsg
 			for seq, lseq := atomic.LoadUint64(&mb.first.seq), atomic.LoadUint64(&mb.last.seq); seq <= lseq; seq++ {
+				if mb.dmap.Exists(seq) {
+					// Optimisation to avoid calling cacheLookup which hits time.Now().
+					updateLLTS = true
+					continue
+				}
 				sm, _ := mb.cacheLookup(seq, &smv)
 				if sm == nil || sm.subj == _EMPTY_ || !lbm[sm.subj] {
 					continue
 				}
+				updateLLTS = false // cacheLookup already updated it.
```

Comment:
Since we are looping here we probably need a llts updated to turn it off. Since with this if we have interleaved records that are there and deleted but do not match we will set it for each lookup and again for the deletes.

Reply:
True, should we expand cacheLookup to call cacheLookupEx that takes a bool indicating if we should update LLTS?

Reply:
We could do, would you prefer I take that approach for this PR?

Reply:
Not here but wanted the context here if we decide to do new PR.